neo/Network/P2P/Message.cs

```diff
@@ -102,6 +102,8 @@ void ISerializable.Deserialize(BinaryReader reader)
         {
             Flags = (MessageFlags)reader.ReadByte();
             Command = (MessageCommand)reader.ReadByte();
+            if (!Enum.IsDefined(typeof(MessageCommand), Command))
```

Comment:
We don't need to check it here. If we add new command in the future, it can be compatible with the current version.

Reply:
The idea of strict enum checking is to fail early. Without this check here I can create a message with a fake command that 
1. the system will first try to decompress
2. next all `IP2PPlugin`s will receive it 
3. and finally it will silently be ignored because there is no matching case in the protocolhandler. 

Instead with this check it will never waste any resources on decompressing and prevents it reaching `P2PPlugin`s. It will drop the remote node on the first invalid message format, which is the same behaviour we have when a payload inside the message is invalid.

Reply:
That is because we don't know the message format, but maybe the `IP2PPlugin` may know it. This could be a protocol extension plugin.